### ISSUE TITLE
Unable to download dependencies from maven

### ISSUE THREAD

**Issue by AmanPandey0320** [micrometer-metrics/context-propagation#268]
The problem occurred while build context-propagation from source. The error message is as follows
```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'context-propagation'.
> Could not resolve all files for configuration ':classpath'.
> Could not find org.ajoberstar.grgit:grgit-core:4.0.2.
Searched in the following locations:
- https://repo.maven.apache.org/maven2/org/ajoberstar/grgit/grgit-core/4.0.2/grgit-core-4.0.2.pom
- https://plugins.gradle.org/m2/org/ajoberstar/grgit/grgit-core/4.0.2/grgit-core-4.0.2.pom
Required by:
project : > com.netflix.nebula:nebula-release-plugin:16.1.0

* Try:
> Run with --info or --debug option to get more log output.
> Get more help at https://help.gradle.org.
```

Can you suggest any workaround to build context-propagation from source?

**Comment by jonatan-ivanov** [comment]
I've edited your comment to improve the formatting. You might want to check out this [Mastering Markdown guide](https://guides.github.com/features/mastering-markdown/) for future reference.

Is this the `main` branch? Can you do a `git pull`?
We are using a never version of the publishing plugin which does not even use grgit anymore:, see [`libs.versions.toml`](https://github.com/micrometer-metrics/context-propagation/blob/06b2769f2606404a94a0c6a38f5548287667de93/gradle/libs.versions.toml#L22C1-L22C105):
```toml
plugin-nebulaPublishing = { module = "com.netflix.nebula:nebula-publishing-plugin", version = "20.3.0" }
```

**Comment by jonatan-ivanov** [comment]
Let me close the issue since we should not use grgit, please let us know if a simple `git pull` does not solve this and we can reopen.

**Comment by AmanPandey0320** [comment]
It is actually required by `nebula-release-plugin:16.1.0` in the 1.0.6 version of context propagation and we require that specific for our use case

**Comment by shakuzen** [comment]
Unfortunately, it looks like that version of grgit was lost with Bintray. See [this note](https://github.com/ajoberstar/grgit?tab=readme-ov-file#old-versions-from-bintrayjcenter) in the grgit README. You could try adding the repository mentioned there if you really need to build context-propagation `1.0.6` from source rather than consume the [release artifact](https://central.sonatype.com/artifact/io.micrometer/context-propagation/1.0.6) we published to Maven Central.

**Comment by jonatan-ivanov** [comment]
I see, there is no way we can fix `v1.0.6` for you since that version was released. I think you have three options though:

1. Add the "backup" repo as Tommy suggested above.
2. Bump up the version number of the release plugin to `18.0.8`, the upgrade path seems simple, see https://github.com/micrometer-metrics/context-propagation/commit/1933afbdc139c0f702a55864cd05d71ea8f954f9 and https://github.com/micrometer-metrics/context-propagation/commit/cbe2eae57a8619874734986fe00b4af0af260f45.
3. Use `1.1.1`, this should be a relatively safe upgrade, if you look at the [diff](https://github.com/micrometer-metrics/context-propagation/compare/v1.0.6...v1.1.1), only new things were added and test/build dependencies were upgraded, existing behavior was not changed or removed.

I would recommend going with `#3` can that work for you?

**Comment by AmanPandey0320** [comment]
Thanks @jonatan-ivanov we will proceed with `#3`.
